compose/ui/ui/src/uikitInstrumentedTest/kotlin/android/compose/ui/test/UIKitInstrumentedTest.kt

```diff
@@ -0,0 +1,200 @@
+        )
+    }
+
+    fun showKeyboard(height: Dp = 216.dp, animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillShowNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(height, animated)
+            )
+        )
+        keyboardHeight = height
+        resizeKeyboard(height, animated)
+    }
+
+    fun resizeKeyboard(newHeight: Dp, animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillChangeFrameNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(newHeight, animated)
+            )
+        )
+        keyboardHeight = newHeight
+    }
+
+    fun hideKeyboard(animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillChangeFrameNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(0.dp, animated)
+            )
+        )
+        keyboardHeight = 0.dp
+    }
+
+    private val isIdle: Boolean get() {
+        val hadSnapshotChanges = Snapshot.current.hasPendingChanges()
+        val hasPendingRecomposer = Recomposer.runningRecomposers.value.any {
+            it.hasPendingWork
+        }
+        val isApplyObserverNotificationPending = Snapshot.isApplyObserverNotificationPending
+
+        return !hadSnapshotChanges && !hasPendingRecomposer && !isApplyObserverNotificationPending
+    }
+
+    @OptIn(ExperimentalNativeApi::class)
+    fun waitForIdle(deadline: Duration = 5.seconds): Boolean {
+        val defaultIdleScore = 10
+        var idleScore = defaultIdleScore
+        val timeoutAchieved = waitForExpectation(deadline = deadline) {
+            if (isIdle) {
+                idleScore -= 1
+            } else {
+                idleScore = defaultIdleScore
+            }
+            idleScore == 0
+        }
+        assert(timeoutAchieved) { "waitForIdle reached $deadline deadline" }
+        return timeoutAchieved
+    }
+
+    fun waitForExpectation(deadline: Duration = 5.seconds, expectation: () -> Boolean): Boolean {
+        val runLoop = NSRunLoop.currentRunLoop()
+        var remainingTime = deadline
+        var time = NSDate.timeIntervalSinceReferenceDate()
+        while (!expectation() && remainingTime > 0.seconds) {
+            runLoop.runUntilDate(NSDate.dateWithTimeIntervalSinceNow(0.005))
+            val current = NSDate.timeIntervalSinceReferenceDate()
+            remainingTime -= (current - time).seconds
+            time = current
+        }
+        return remainingTime > 0.seconds
+    }
+
+    fun runTest(block: suspend UIKitInstrumentedTest.() -> Unit) {
+        runBlocking {
+            block()
+            hideKeyboard(animated = false)
+        }
+    }
+}
```

Comment:
I think this does too much for a generic `runTest` function:
1. Why force the test block to be `suspend`?
2. If it does need to be `suspend`, why force running it with `runBlocking`? We could just make `runTest` suspend.
3. Hiding the keyboard seems too specific. We can't anticipate all scenarios where "cleanup" after the test is needed. The user should clean up after their test themselves.

Reply:
1. `suspend` environment will help us to test various scenarios with delay. For me it's a good improvement of the tests runner.
2. runTest cannot be suspend because of `'suspend' functions annotated with @kotlin.test.Test are unsupported`.
3. Ok, removed.

Reply:
Done